turbopack/crates/turbo-rcstr/src/lib.rs

```diff
@@ -276,13 +298,18 @@ impl Default for RcStr {
 
 impl PartialEq for RcStr {
     fn eq(&self, other: &Self) -> bool {
-        match (self.tag(), other.tag()) {
-            (DYNAMIC_TAG, DYNAMIC_TAG) => {
+        // For inline RcStrs this is sufficient and for out of line values it handles a simple
+        // identity cases
+        if self.unsafe_data == other.unsafe_data {
```

Comment:
This adds a fast path for identical ptrs that wasn't there before. I wonder why this didn't exist before, but it should be a low hanging performance win too

Reply:
added to the PR description

prior to the 'prehashed' optimisation this impl was just
```
impl PartialEq for RcStr {
    fn eq(&self, other: &Self) -> bool {
        self.as_str() == other.as_str()
    }
}
```
so idk, the implementation was so simple and obvious that the optimization wasn't obvious.  I should backport this to swc too!